go/src/koding/klient/machine/dynamic.go

```diff
@@ -192,10 +192,14 @@ func (dc *DynamicClient) cron() {
 
 // tryUpdate uses client builder to ping the machine and updates dynamic client
 // if machine address changes.
-func (dc *DynamicClient) tryUpdate(addr *Addr) {
+func (dc *DynamicClient) tryUpdate(addr *Addr, islog *bool) {
 	stat, a, err := dc.opts.Builder.Ping(dc.opts.AddrFunc)
 	if err != nil {
-		dc.log.Warning("Machine ping error: %s", err)
+		if *islog {
```

Comment:
> The code you linked also irrelevant to your proposal. It is an exposed function.

I was tricked by golang URLs format. And changed:
https://golang.org/src/net/http/client.go?s=6486:6538#L189
to
https://golang.org/src/net/http/client.go?s=6486:6538#L209
but it still links to line 189. Should have checked this.

I meant line [209](https://golang.org/src/net/http/client.go?#L209), where req is not checked for `nil`.

You are right, pointing to someone's code is generally destructive. Unless it's pointing to code of people you respect as deveopers - I pointed yours and these written by go authors:)